### PR TITLE
try to clean up workflow issue

### PR DESCRIPTION
**Pull request by bricepollock**
### Description
There was a workflow syntax error, hopefully this fixes it.

### Test Plan
* This PR

### CODE REVIEW

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -1,24 +1,25 @@
...
 jobs:
   test:
     runs-on: macOS-latest
+    
     steps:
-	    # Checkout the code to the working directory
-      - uses: actions/checkout@v2
-      # Install xcbeautify so the logs are human-friendly
-      - name: install xcbeautify
+      - uses: actions/checkout@v2 #checkout working dir
```
Comment (17jmumford) [inline .github/workflows/main.yml:13]: v4 of checkout released this year. It has improved stability, logging, and security, but is functionally the same.
```suggestion
      - uses: actions/checkout@v4 #checkout working dir
```

Comment (17jmumford) [inline .github/workflows/main.yml:13]:
```suggestion
      - uses: actions/checkout@v4 #checkout working dir
```
Recommend v4 for increased stability, security, and improved logging.

**Review by 17jmumford — APPROVED** [review]

File: .github/workflows/main.yml (hunk 1 above)
Comment (17jmumford) [reply .github/workflows/main.yml:13]: functionally it is the same

**Comment by 17jmumford** [comment]
I approved the change, but it looks like the xcode test step is taking an abnormally long time. You could try running the tests locally and see if a process is hanging or something is stuck in a loop.

**Comment by 17jmumford** [comment]
The xcode test step is also not printing log statements. You could try adding a -verbose flag to xcbeautify, or removing it.

**Comment by bricepollock** [comment]
@17jmumford if by abnormally long you mean 10 minutes, that is very normal for macOS / Xcode tasks on CI. Its actually quite fast in our world.

**Comment by bricepollock** [comment]
All Green! CI is reporting failures. Ship it!

**Pull request merged**